src/coreclr/jit/valuenum.cpp

```diff
@@ -7813,7 +7813,104 @@ ValueNum ValueNumStore::EvalHWIntrinsicFunBinary(var_types      type,
     }
     return VNForFunc(type, func, arg0VN, arg1VN);
 }
+
+ValueNum EvaluateSimdFloatWithElement(
+    ValueNumStore* vns, var_types type, var_types baseType, ValueNum arg0VN, int index, float value)
+{
+    assert(vns->IsVNConstant(arg0VN));
+
+    switch (type)
+    {
+        case TYP_SIMD8:
+        {
+            simd8_t cnsVec    = vns->GetConstantSimd8(arg0VN);
+            cnsVec.f32[index] = value;
```

Comment:
your code seems to not checking `index` being out of bounds

Reply:
I thought it shouldn't be necessary because gtNewSimdWithElementNode asserts that:
https://github.com/dotnet/runtime/blob/362678a6c7d45f9a49c4aaa946701f791c56b484/src/coreclr/jit/gentree.cpp#L24590-L24602

Added an assert. Do you think it's sufficient?

Reply:
You need to explicitly check the bounds, otherwise you might get an AV in JIT time